### PR TITLE
add new rule audit_rules_immutable_login_uids

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- add new rule including OVAL, Bash, Ansible
- add rule to RHEL8 STIG profile

#### Rationale:

- the rule checks for the only line "--loginuid-immutable" compared to rule audit_immutable_login_uids
- It is related to this item: https://stigaview.com/products/rhel8/v1r8/RHEL-08-030122/

- Fixes # _Issue number here (e.g. #26) or remove this line if no issue exists._

#### Review Hints:

- _Review hints here. Replace this text. Don't use the italics format!_

- _Use this optional section to give any relevant information which could help the reviewer to more quickly and assertively understand and test the changes._

- _Good examples are useful commands, if it is better to review all commits together or in a suggested sequence, any relevant discussion in other PRs or issues, etc._

### CODE REVIEW

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable_login_uids/bash/shared.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# in case auditctl is used
+if grep -q '^\s*ExecStartPost=-/sbin/auditctl' /usr/lib/systemd/system/auditd.service; then
+  if ! grep -q '^\s*--loginuid-immutable\s*$' /etc/audit/audit.rules; then
+    echo "--loginuid-immutable" >> /etc/audit/audit.rules
+  fi
+else
+  immutable_found=0
+  for f in /etc/audit/rules.d/*.rules; do
+    if grep -q '^\s*--loginuid-immutable\s*$' $f; then
+  immutable_found=1
```
Comment (jan-cerny) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable_login_uids/bash/shared.sh:12]: indentation

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable_login_uids/bash/shared.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# platform = multi_platform_all
+
+# in case auditctl is used
+if grep -q '^\s*ExecStartPost=-/sbin/auditctl' /usr/lib/systemd/system/auditd.service; then
+  if ! grep -q '^\s*--loginuid-immutable\s*$' /etc/audit/audit.rules; then
+    echo "--loginuid-immutable" >> /etc/audit/audit.rules
+  fi
+else
+  immutable_found=0
+  for f in /etc/audit/rules.d/*.rules; do
```
Comment (jan-cerny) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable_login_uids/bash/shared.sh:10]: what if there is no .rules file in this directory?

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable_login_uids/bash/shared.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    if grep -q '^\s*--loginuid-immutable\s*$' $f; then
```
Comment (jan-cerny) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable_login_uids/bash/shared.sh:11]: quote $f

**Comment by vojtapolasek** [comment]
@jan-cerny  feedback addressed in https://github.com/ComplianceAsCode/content/pull/10070/commits/2b8bf7e7df11bfc0eaf4370ab8a77f0fac47d942

**Review by jan-cerny — APPROVED** [review]
I have checked that the supplied test scenarios have passed in the Automatus CS8 job in CI.

**Pull request merged**